src/sonic-eventd/src/eventd.cpp

```diff
@@ -409,8 +409,30 @@ capture_service::do_capture()
          *
          * This behavior will only happen once when XSUB connects to XPUB not everytime cache is started.
          *
+         * There are chances that there are events already published to XSUB endpoint before XSUB is able to connect to XPUB, so we can receive events
+           before the subscription message
          */
-         init_done = true;
+
+
+        if(rc == 1) { // Expected case to receive subscription message as very first message
+            SWSS_LOG_INFO("Received subscription message when XSUB connects to XPUB");
+        } else if (rc > 1) { // If there are events already published to XSUB before XSUB connects to XPUB, we can receive events before subscription message
+            string event_source((const char*)zmq_msg_data(&msg), zmq_msg_size(&msg));
+            SWSS_LOG_DEBUG("Receiving event from source: %s, will read second part of event", event_source.c_str());
+            int more = 0;
+            size_t more_size = sizeof(more);
+            zmq_getsockopt(cap_sub_sock, ZMQ_RCVMORE, &more, &more_size);
+            if(more) {
```

Comment:
Do we need a loop here to read all message?

Also, if there is an ADO please update the PR description.

Reply:
Changed logic to not account for capture service capturing control character. Will be done in zmq_read_part

https://github.com/sonic-net/sonic-swss-common/pull/906